### ISSUE TITLE
Adding gRPC instrumentation but not HttpClient causes issues with parenting

### ISSUE THREAD

**Issue by alanwest** [open-telemetry/opentelemetry-dotnet#1305]
Both the gRPC .NET and HttpClient libraries generate their own Activity for outgoing calls.

gRPC .NET uses HttpClient behind the scenes, so when you configure OpenTelemetry with both gRPC and HttpClient instrumentation you'll see that the Activity generated by gRPC is the parent of the Activity generated by HttpClient.

However, if you only include gRPC instrumentation or if you only include HttpClient instrumentation the chain of parenting can become broken since the library still generates their respective Activity. Another way this can occur is if you add both gRPC and HttpClient instrumentation, but you configure `SuppressDownstreamInstrumentation` for the gRPC instrumentation thereby excluding the child HttpClient Activity from the trace.

This issue is somewhat related to #1217 in that the general issue is that libraries generate Activities regardless of whether OpenTelemetry has been configured to be interested in them.

**Comment by cijothomas** [comment]
https://github.com/open-telemetry/opentelemetry-dotnet/pull/1464 This PR should address this, right ? atleast partially?

**Comment by alanwest** [comment]
> #1464 This PR should address this, right ? atleast partially?

Yes, correct parenting will be maintained when using `SuppressDownstreamInstrumentation`. I think it's ok to declare this "not a bug" and just document this behavior.

**Comment by martinjt** [comment]
Is this still something we have to do @alanwest ? it's been 3 years and I've not seen any issues with regards to this.
